src/lib/ndpi_utils.c

```diff
@@ -2433,3 +2433,36 @@ void ndpi_set_tls_cert_expire_days(struct ndpi_detection_module_struct *ndpi_str
 				   u_int8_t num_days) {
   ndpi_str->tls_certificate_expire_in_x_days = num_days;
 }
+
+/* ******************************************* */
+
+int ndpi_vsnprintf(char * str, size_t size, char const * format, va_list va_args)
+{
+#ifdef WIN32
+  if (str == NULL || size == 0 || format == NULL)
+  {
+    return -1;
+  }
+
+  int ret = vsnprintf_s(str, size, _TRUNCATE, format, va_args);
```

Comment:
The issue here is a Windoze one. The Windows c library does not obey C99. `snprintf` does not guarantee a `NUL` terminated string if the output buffer is not big enough for the formatted string. That results in test diff's e.g. for `tests/pcap/1kxun.pcap`.

I thought about using the Mingw-w64 builtin, but decided against it, because that will also reduce portability for e.g. MSVC (if we ever want to do that).